src/main/java/org/sonar/plugins/stash/StashIssueReportingPostJob.java

```diff
@@ -54,55 +54,81 @@ public void executeOn(Project project, SensorContext context) {
+                                                                  stashCredentials.getLogin(), stashClient);
+        }
+        
+        // if threshold exceeded, do not push issue list to Stash
+        if (issueReport.countIssues() >= issueThreshold) {
+          LOGGER.warn("Too many issues detected ({}/{}): Issues cannot be displayed in Diff view",
+                                                         issueReport.countIssues(), issueThreshold);
+        } else {
+          stashRequestFacade.postCommentPerIssue(stashProject, repository, stashPullRequestId, sonarQubeURL,
+                                                               issueReport, diffReport, stashClient);
+        }
 
-            stashRequestFacade.postAnalysisOverview(stashProject, repository, stashPullRequestId, sonarQubeURL, issueThreshold, issueReport, stashClient);
-           
-            if (canApprovePullrequest) {
-           
-              // if no new issues, plugin approves the pull-request 
-              if (issueReport.countIssues() == 0) {
-                stashRequestFacade.approvePullRequest(stashProject, repository, stashPullRequestId, stashCredentials.getLogin(), stashClient);
-              } else {
-                stashRequestFacade.resetPullRequestApproval(stashProject, repository, stashPullRequestId, stashCredentials.getLogin(), stashClient);
-              }
-            }
+        stashRequestFacade.postAnalysisOverview(stashProject, repository, stashPullRequestId, sonarQubeURL,
+                                                              issueThreshold, issueReport, stashClient);
+       
+        if (canApprovePullrequest) {
+       
+          // if no new issues, plugin approves the pull-request 
+          if (issueReport.countIssues() == 0) {
+            stashRequestFacade.approvePullRequest(stashProject, repository, stashPullRequestId,
+                                                                stashCredentials.getLogin(), stashClient);
+          } else {
+            stashRequestFacade.resetPullRequestApproval(stashProject, repository, stashPullRequestId,
+                                                                      stashCredentials.getLogin(), stashClient);
           }
         }
       }
     } catch (StashConfigurationException e) {
       LOGGER.error("Unable to push SonarQube report to Stash: {}", e.getMessage());
       LOGGER.debug("Exception stack trace", e);
+
+    } catch (StashMissingElementException e) {
+      LOGGER.error("Process stopped: {}", e.getMessage());
+      LOGGER.debug("Exception stack trace", e);
+
     } finally {
       if (stashClient != null) {
         stashClient.close();
       }
     }
   }
+
+  /*
+     Custom exception to keep nested if statements under control
+  */
+  private class StashMissingElementException extends Exception {
+
+    public StashMissingElementException(String exc) {
+        super(exc);
+    }
+    public String getMessage() {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1161)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this method to simply inherit it. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1185)